src/snovault/elasticsearch/indexer.py

```diff
@@ -285,41 +458,124 @@ def index(request):
                         item['error_message'] = "Error occured during indexing, check the logs"
                 result['errors'] = error_messages
 
 
-        es.indices.refresh(RESOURCES_INDEX)
+        request.registry[ELASTIC_SEARCH].indices.refresh(RESOURCES_INDEX)
         if flush:
             try:
-                es.indices.flush_synced(index=RESOURCES_INDEX)  # Faster recovery on ES restart
+                request.registry[ELASTIC_SEARCH].indices.flush_synced(index=RESOURCES_INDEX)  # Faster recovery on ES restart
             except ConflictError:
                 pass
+        return result, indexing_update_infos
 
+    # Load indexing
+    output_tuple = _load_indexing(request, session, connection, indexer_state)
+    result, invalidated, flush, first_txn, snapshot_id, restart, xmin, return_now = output_tuple
+    if return_now:
+        return result
+    indexing_update_infos = []
+    dry_run = request.json.get('dry_run', False)
+    if invalidated and not dry_run:
+        run_indexing_vars = (
+            request,
+            indexer_state,
+            result,
+            invalidated,
+            stage_for_followup,
+            flush,
+            snapshot_id,
+            restart,
+            xmin,
+        )
+        if this_node:
+            if this_node['node_index'] == HEAD_NODE_INDEX:
+                # Check remote indexing
+                uuids_count = len(invalidated)
+                do_remote_indexing = _determine_indexing_protocol(request, uuids_count)
+                if did_timeout:
+                    log.warning(f"Remote indexing index: did_timeout.  Run on head node.")
+                    head_node, indexing_node, time_now, did_fail, is_indexing_node  = setup_indexing_nodes(
+                        request,
+                        indexer_state,
+                        reset=True
+                    )
+                    result, indexing_update_infos = _run_indexing(*run_indexing_vars)
+                elif do_remote_indexing:
+                    log.warning(f"Remote indexing run indexing: Defer to remote.")
+                    instance_name = this_node['instance_name']
+                    status_code = _send_sqs_msg(
+                        f"{instance_name}-cluster",
+                        remote_indexing,
+                        'Turn ON remote indexer',
+                        delay_seconds=60,
+                        save_event=False
+                    )
+                    if status_code == 200:
+                        this_node['waiting_on_remote'] = True
+                        this_node['last_run_time'] = str(float(time.time()))
+                        indexer_state.put_obj(this_node['node_index'], this_node)
+                    else:
+                        log.warning(f"Remote indexing index: Failed sqs queue.  Run on head node.")
+                        result, indexing_update_infos = _run_indexing(*run_indexing_vars)
```

Comment:
not a fan of * here.  It's somewhat obfuscatory.   Probably I would just pas a dict around but maybe there is a good reason not to do so.